src/main/groovy/com/morpheus/scvmm/sync/VirtualMachineSync.groovy

```diff
@@ -0,0 +1,490 @@
+                cloud            : cloud,
+                status           : 'provisioned',
+                apiKey           : java.util.UUID.randomUUID(),
+                account          : cloud.account,
+                managed          : false,
+                uniqueId         : cloudItem.ID,
+                provision        : false,
+                hotResize        : false,
+                serverType       : 'vm',
+                lvmEnabled       : false,
+                discovered       : true,
+                internalId       : cloudItem.VMId,
+                externalId       : cloudItem.ID,
+                displayName      : cloudItem.Name,
+                singleTenant     : true,
+                computeServerType: defaultServerType,
+                powerState       : cloudItem.VirtualMachineState == 'Running' ? ComputeServer.PowerState.on : ComputeServer.PowerState.off
+        ]
+        return vmConfig
+    }
+
+    def syncVolumes(server, externalVolumes) {
+        log.debug "syncVolumes: ${server}, ${groovy.json.JsonOutput.prettyPrint(externalVolumes?.encodeAsJSON()?.toString())}"
+        def changes = false //returns if there are changes to be saved
+        try {
+            def maxStorage = 0
+
+            def existingVolumes = server.volumes
+            def masterItems = externalVolumes
+
+            def existingItems = Observable.fromIterable(existingVolumes)
+            def diskNumber = masterItems.size()
+
+            SyncTask<StorageVolumeIdentityProjection, Map, StorageVolume> syncTask = new SyncTask<>(existingItems, masterItems as Collection<Map>)
+
+            syncTask.addMatchFunction { StorageVolumeIdentityProjection storageVolume, Map masterItem ->
+                storageVolume.externalId == masterItem.ID
+            }.withLoadObjectDetailsFromFinder { List<SyncTask.UpdateItemDto<StorageVolumeIdentityProjection, StorageVolume>> updateItems ->
+                context.async.storageVolume.listById(updateItems.collect { it.existingItem.id } as List<Long>)
+            }.onAdd { itemsToAdd ->
+                addMissingStorageVolumes(itemsToAdd, server, diskNumber, maxStorage, changes)
+            }.onUpdate { List<SyncTask.UpdateItem<StorageVolume, Map>> updateItems ->
+                updateMatchedStorageVolumes(updateItems, server, maxStorage, changes)
+            }.onDelete { removeItems ->
+                removeMissingStorageVolumes(removeItems, server, changes)
+            }.start()
+
+            if (server instanceof ComputeServer && server.maxStorage != maxStorage) {
+                log.debug "max storage changed for ${server} from ${server.maxStorage} to ${maxStorage}"
+                server.maxStorage = maxStorage
+                context.async.computeServer.save(server).blockingGet()
+                changes = true
+            }
+        } catch (e) {
+            log.error("syncVolumes error: ${e}", e)
+        }
+        return changes
```

Comment:
returns changes so you could use the return value to determine if the affected item needs save and build a save list to do a build save in the parent method.